src/main/java/org/example/repository/CrudRepository.java

```diff
@@ -1,12 +1,24 @@
 package org.example.repository;
 
-public interface CrudRepository <T> {
+import org.example.entity.Order;
+import org.example.entity.Product;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface CrudRepository<T> {
 
     T findById(int id);
 
     T create(T entity);
 
     void delete(int id);
 
-    T update (T entity);
+    T update(T entity);
+
+    Optional<Order> getOrderById(int id);
+
+    List<Order> getOrdersByCustomer(int customerId);
+
+    Optional<Product> getProductById(int productId);
```

Comment:
No, you can't add those here, it's a generic interface. Specific methods for specific entities must be in their respective repositories. This one is a common repository. Please remove